zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -549,6 +549,7 @@ public static Authentication getAuthentication() {
                 .toList());
         if (authenticatedPrincipal instanceof final CamundaUser user) {
           authenticatedUserKey = user.getUserKey();
+          claims.put(Authorization.AUTHORIZED_USERNAME, user.getUsername());
```

Comment:
❓ Do we need to add a `null` check to this line, like we had before? Or is it guaranteed that the `username` is always set in the `CamundaUser`?

Reply:
I'll always be set. We fetch the user from ES and create this `CamundaUser` [here](https://github.com/camunda/camunda/blob/main/authentication/src/main/java/io/camunda/authentication/CamundaUserDetailsService.java#L66-L75). And username is a required field.

Reply:
Regarding the null-check before. We had it on the `auhtenticatedUserKey`. We set that only in 1 place. So I moved putting this claim there. We only need it for basic auth and with basic auth we will always have the user.